src/dsp_tools/commands/xml_validate/prepare_input.py

```diff
@@ -0,0 +1,150 @@
+from pathlib import Path
+from typing import cast
+
+from lxml import etree
+
+from dsp_tools.commands.xml_validate.models.deserialised import LinkValueDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import ListValueDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import PermissionsDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import ProjectDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import ResourceDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import RichtextDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import SimpleTextDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import ValueDeserialised
+from dsp_tools.commands.xml_validate.models.xml_deserialised import PermissionsXML
+from dsp_tools.commands.xml_validate.models.xml_deserialised import ProjectXML
+from dsp_tools.commands.xml_validate.models.xml_deserialised import ResourceXML
+from dsp_tools.utils.xml_utils import parse_and_clean_xml_file
+from dsp_tools.utils.xml_validation import validate_xml
+
+
+def parse_file(file: Path) -> ProjectXML:
+    """Returns an object which follows the structure of the XML closely"""
+    root = _parse_file_validate_with_schema(file)
+    return _transform_into_xml_deserialised(root)
+
+
+def _parse_file_validate_with_schema(file: Path) -> etree._Element:
+    root = parse_and_clean_xml_file(file)
+    validate_xml(root)
+    return root
+
+
+def _transform_into_xml_deserialised(root: etree._Element) -> ProjectXML:
+    shortcode = root.attrib["shortcode"]
+    default_ontology = root.attrib["default-ontology"]
+    permissions = [_get_permissions(x) for x in root.iterdescendants(tag="permissions")]
```

Comment:
are permissions ever relevant in the experiment setup? if not, I'd simply drop them entirely from the model and here from the deserialisation

Reply:
I want to include them in the validation. No permissions should be entered that are not specified in the XML. But since I did not plan on violating that in the test data I can remove it. What do you think?